src/System.IO.Pipelines/System/IO/Pipelines/ReadCursor.cs

```diff
@@ -235,24 +293,17 @@ private bool TryGetBufferMultiBlock(ReadCursor end, out Memory<byte> data)
                 else
                 {
                     segment = segment.Next;
-                    index = segment.Start;
+                    startIndex = segment.Start;
                 }
             }
 
-            data = segment.Buffer.Slice(index, following);
+            data = segment.Memory.Slice(startIndex, following);
             return true;
         }
 
         public override string ToString()
         {
-            if (IsEnd)
-            {
-                return "<end>";
-            }
-
             var sb = new StringBuilder();
-            Span<byte> span = Segment.Buffer.Span.Slice(Index, Segment.End - Index);
-            SpanLiteralExtensions.AppendAsLiteral(span, sb);
             return sb.ToString();
```

Comment:
This could be optimized to `return "";` 😆

Reply:
Removed the method entirely.